sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -216,6 +208,7 @@ public async Task ProcessEventsAsync(ProcessorPartitionContext context, IEnumera
                 // code, and capture/log/persist failed events, since they won't be retried.
                 if (messages.Any())
```

Comment:
You may want to consider capturing the `messages.ToArray()` return from line 153 and using that here; you're enumerating an `IEnumerable<T>` multiple times.   Pragmatically, we return `IReadOnlyList<EventData>` from the `EventProcessor<T>`.    Since you're already paying the array transform cost,  may as well use it.     I wonder if, long term, it would make sense to use `IReadOnlyList` instead of an array and avoid the transforms altogether.